include/IVideoDriver.h

```diff
@@ -231,15 +231,15 @@ namespace video
 		virtual void drawMeshBuffer(const video::IGPUMeshBuffer* mb) =0;
 
 		//! Indirect Draw
-		virtual void drawArraysIndirect(const asset::IMeshDataFormatDesc<video::IGPUBuffer>* vao,
-                                        const asset::E_PRIMITIVE_TYPE& mode,
+		virtual void drawArraysIndirect(const IGPUMeshBuffer::SBufferBinding _vtxBindings[IGPUMeshBuffer::MAX_ATTR_BUF_BINDING_COUNT],
+                                        asset::E_PRIMITIVE_TOPOLOGY mode,
                                         const IGPUBuffer* indirectDrawBuff,
-                                        const size_t& offset, const size_t& count, const size_t& stride) =0;
-		virtual void drawIndexedIndirect(const asset::IMeshDataFormatDesc<video::IGPUBuffer>* vao,
-                                        const asset::E_PRIMITIVE_TYPE& mode,
-                                        const asset::E_INDEX_TYPE& type,
+                                        size_t offset, size_t count, size_t stride) = 0;
+		virtual void drawIndexedIndirect(const IGPUMeshBuffer::SBufferBinding _vtxBindings[IGPUMeshBuffer::MAX_ATTR_BUF_BINDING_COUNT],
+                                        asset::E_PRIMITIVE_TOPOLOGY mode,
+                                        asset::E_INDEX_TYPE indexType, const IGPUBuffer* indexBuff,
                                         const IGPUBuffer* indirectDrawBuff,
-                                        const size_t& offset, const size_t& count, const size_t& stride) =0;
+                                        size_t offset, size_t count, size_t stride) = 0;
```

Comment:
you want to add two last parameters `const IGPUBuffer* countBuffer=nullptr, size_t countOffset=0u`, also rename `count` to `maxCount`.

if the parameter buffer if its not null and (GL_ARB_indirect_parameters || Version>=460 ):
1. attach (via state chace) to the indirect parameter binding
2. MultiDraw{Arrays|Elements}IndirectCount variants instead of the usual MDI call

else if parameter buffer not null, scream to high heaven and ouput log error.